server/src/com/thoughtworks/go/config/update/RoleConfigDeleteCommand.java

```diff
@@ -38,11 +38,16 @@ public void update(CruiseConfig preprocessedConfig) throws Exception {
         }
 
         removeFromServerRole(preprocessedConfig, preprocessedRole);
+        removeFromAdmins(preprocessedConfig);
```

Comment:
We haven't released api so there was no such bad configuration. This will only happens if user try to delete role from api.